provider/hetzner/hetzner.go

```diff
@@ -170,24 +171,30 @@ func (p *HetznerProvider) submitChanges(ctx context.Context, changes []*HetznerC
 					TTL:        change.ResourceRecordSet.TTL,
 				}
 				answer, err := p.Client.CreateRecord(record)
-				if err != nil {
+				if err != nil || answer.Error.Code != 0 {
 					log.WithFields(log.Fields{
 						"Code":         answer.Error.Code,
 						"Message":      answer.Error.Message,
 						"Record name":  answer.Record.Name,
 						"Record type":  answer.Record.RecordType,
 						"Record value": answer.Record.Value,
 					}).Warning("Create problem")
-					return err
+					if err != nil {
+						return err
+					}
+					return fmt.Errorf("could not create record %s (%d)", change.ResourceRecordSet.Name, answer.Error.Code)
```

Comment:
We could do something along the lines of:

```go
if err != nil {
     return err
} 

if answer.Error.Code != 0 {
	log.WithFields(log.Fields{
		"Code":         answer.Error.Code,
		"Message":      answer.Error.Message,
		"Record name":  answer.Record.Name,
		"Record type":  answer.Record.RecordType,
		"Record value": answer.Record.Value,
	}).Warning("Create problem")
	return fmt.Errorf("could not create record %s (%d)", change.ResourceRecordSet.Name, answer.Error.Code)
```

I think if err != nil, then the answer should be nil according to Go convention. Anyway even if the hetzner provider doesn't do this well, it is still better to avoid nested `if err`.